### ISSUE TITLE
kmsg: Logging issue or crash ?

### ISSUE THREAD

**Issue by epcim** [fluent/fluent-bit#392]
case:
When on load, seems to crash as sending kernel msgs. (btw lack of support for position file in kmsg input plugin)

Note, it may be also the case kmsg fails to send events || error from another plugin/workflow step (but we don't know which one), etc..

update: the kmsg was working fine, until I enabled all the input plugins

```
Oct 09 18:32:04 d2c-76-8a-ac-64-6e td-agent-bit[18386]: [2017/10/09 18:32:04] [debug] [input kmsg.0] [mem buf] size = 118718
Oct 09 18:32:04 d2c-76-8a-ac-64-6e td-agent-bit[18386]: [2017/10/09 18:32:04] [debug] [input kmsg.0] [mem buf] size = 118876
Oct 09 18:32:04 d2c-76-8a-ac-64-6e td-agent-bit[18386]: [2017/10/09 18:32:04] [debug] [input kmsg.0] [mem buf] size = 119069
Oct 09 18:32:04 d2c-76-8a-ac-64-6e td-agent-bit[18386]: [2017/10/09 18:32:04] [debug] [input kmsg.0] [mem buf] size = 119262
Oct 09 18:32:04 d2c-76-8a-ac-64-6e td-agent-bit[18386]: [2017/10/09 18:32:04] [debug] [input kmsg.0] [mem buf] size = 119492
Oct 09 18:32:04 d2c-76-8a-ac-64-6e td-agent-bit[18386]: [2017/10/09 18:32:04] [debug] [input kmsg.0] [mem buf] size = 119644
Oct 09 18:32:04 d2c-76-8a-ac-64-6e td-agent-bit[18386]: [2017/10/09 18:32:04] [debug] [input kmsg.0] [mem buf] size = 119811
Oct 09 18:32:14 d2c-76-8a-ac-64-6e systemd[1]: td-agent-bit.service: main process exited, code=killed, status=11/SEGV
Oct 09 18:32:14 d2c-76-8a-ac-64-6e systemd[1]: Unit td-agent-bit.service entered failed state.
Oct 09 18:32:14 d2c-76-8a-ac-64-6e systemd[1]: td-agent-bit.service holdoff time over, scheduling restart.
Oct 09 18:32:14 d2c-76-8a-ac-64-6e systemd[1]: Stopping TD Agent Bit...
Oct 09 18:32:14 d2c-76-8a-ac-64-6e systemd[1]: Starting TD Agent Bit...
Oct 09 18:32:14 d2c-76-8a-ac-64-6e systemd[1]: Started TD Agent Bit.
```

**Comment by epcim** [comment]
another occurrence:

```
Oct 09 18:39:31 d2c-76-8a-ac-64-6e td-agent-bit[30377]: [2017/10/09 18:39:31] [debug] [input tail.1] [mem buf] size = 568
Oct 09 18:39:31 d2c-76-8a-ac-64-6e td-agent-bit[30377]: [2017/10/09 18:39:31] [debug] [in_tail] file=/var/lib/pgsql/data/pg_log/postgresql.log-201710090000 read=246 lines=3
Oct 09 18:39:34 d2c-76-8a-ac-64-6e td-agent-bit[30377]: [2017/10/09 18:39:34] [debug] [in_tail] file=/var/lib/pgsql/data/pg_log/postgresql.log-201710090000 event
Oct 09 18:39:34 d2c-76-8a-ac-64-6e td-agent-bit[30377]: [2017/10/09 18:39:34] [debug] [input tail.1] [mem buf] size = 852
Oct 09 18:39:34 d2c-76-8a-ac-64-6e td-agent-bit[30377]: [2017/10/09 18:39:34] [debug] [in_tail] file=/var/lib/pgsql/data/pg_log/postgresql.log-201710090000 read=246 lines=3
Oct 09 18:39:34 d2c-76-8a-ac-64-6e td-agent-bit[30377]: [2017/10/09 18:39:34] [debug] [in_tail] file=/var/lib/pgsql/data/pg_log/postgresql.log-201710090000 event
Oct 09 18:39:34 d2c-76-8a-ac-64-6e td-agent-bit[30377]: [2017/10/09 18:39:34] [debug] [input tail.1] [mem buf] size = 1136
Oct 09 18:39:34 d2c-76-8a-ac-64-6e td-agent-bit[30377]: [2017/10/09 18:39:34] [debug] [in_tail] file=/var/lib/pgsql/data/pg_log/postgresql.log-201710090000 read=246 lines=3
Oct 09 18:39:36 d2c-76-8a-ac-64-6e td-agent-bit[30377]: [2017/10/09 18:39:35] [debug] [task->buffer] worker_id=0
Oct 09 18:39:36 d2c-76-8a-ac-64-6e td-agent-bit[30377]: [2017/10/09 18:39:35] [debug] [task] created task=0x7f8cde042000 id=0 OK
Oct 09 18:39:36 d2c-76-8a-ac-64-6e systemd[1]: td-agent-bit.service: main process exited, code=killed, status=11/SEGV
Oct 09 18:39:36 d2c-76-8a-ac-64-6e systemd[1]: Unit td-agent-bit.service entered failed state.
Oct 09 18:39:36 d2c-76-8a-ac-64-6e systemd[1]: td-agent-bit.service holdoff time over, scheduling restart.
Oct 09 18:39:36 d2c-76-8a-ac-64-6e systemd[1]: Stopping TD Agent Bit...
Oct 09 18:39:36 d2c-76-8a-ac-64-6e systemd[1]: Starting TD Agent Bit...
Oct 09 18:39:36 d2c-76-8a-ac-64-6e systemd[1]: Started TD Agent Bit.
Oct 09 18:39:36 d2c-76-8a-ac-64-6e td-agent-bit[32566]: [2017/10/09 18:39:36] [ info] [engine] started
```

**Comment by epcim** [comment]
Sorry for chunking this, seems to be related (without multiline tail plugin):

![image](https://user-images.githubusercontent.com/1211552/31353720-ed376b00-ad33-11e7-84e0-a6c4fdf598e1.png)

**Comment by edsiper** [comment]
@epcim 

would you please provide a dump of your /dev/kmsg ?

```
cat /dev/kmsg > kmsg.dump
```

then upload the generated file somewhere.

**Comment by edsiper** [comment]
@epcim ping

**Comment by epcim** [comment]
out of keyboard for 1week for this topic. will ping you once back or coallegues update if stilm an issue. On 11 Oct 2017 17:09, Eduardo Silva <notifications@github.com> wrote:@epcim ping

**Comment by edsiper** [comment]
ping

**Comment by edsiper** [comment]
last ping...

**Comment by epcim** [comment]
sorry for the delay, try this: http://apealive.net/media/kmsg.tar.bz2

**Comment by edsiper** [comment]
I've pushed a fix into git master and 0.12 branch (available on 0.12.7 this week):

https://github.com/fluent/fluent-bit/commit/e93ba300287553c795384f420a85a13466b1b172

**Comment by edsiper** [comment]
@epcim [0.12.7](http://fluentbit.io/announcements/v0.12.7/) has been released.

Closing this issue, if for some reason the problem persist we can reopen it.
